### PR TITLE
Fix send mail

### LINKED ISSUE #13306: Gitea sendmail command fails mysteriously
- Gitea version (or commit ref): [`1.14.0+dev-84-ge2740b32b`](go-gitea/gitea/commit/e2740b32b)
- Git version: `git version 2.29.1.341.ge80a0c044a`
- Operating system: Arch Linux
- Built from source
  - [x] Yes
- Run with
  - [x] systemd
- Database:
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] No

## Description
I have been testing the [recently introduced sendmail functionality](#13079 ) and my instance doesn't want to proceed with the following error
```log
[gitea@nebula ~]$ gitea admin sendmail --title "[Test] Gitea sendmail" --content "Please disregard this email as it is just a test of Gitea sendmail functionality."
Proceed with sending email? [Y/n] y
error: Unable to contact gitea: Post "api/internal/mail/send": unsupported protocol scheme ""
```
The error does not disappear even when I explicitly point the command to a valid config, which does have the data necessary to speak to Gitea and are apparently missing in the POST request being sent out (namely protocol scheme and app URL), as documented below
```log
[gitea@nebula ~]$ gitea admin sendmail --config /etc/gitea/app.ini --title "[Test] Gitea sendmail" --content "Please disregard this email as it is just a test of Gitea sendmail functionality."
Proceed with sending email? [Y/n] y
error: Unable to contact gitea: Post "api/internal/mail/send": unsupported protocol scheme ""
```
Also, logs (set to debug) with gitea in dev mode don't yield anything.

### PR DESCRIPTION
**Pull request by lunny**
Fix #13306

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf — COMMENTED** [review]
I applied this [patch](https://github.com/go-gitea/gitea/pull/13312/commits/168e6a0cf2c5fa05f37d610b904636623b630485.patch) and tried sending the email, here's what I got
```log
[gitea@nebula ~]$ gitea admin sendmail --config /etc/gitea/app.ini --title "[Test00] Gitea sendmail" --content "Please disregard thisemail as it is just a test of Gitea sendmail functionality."
2020/10/26 09:54:39 ...s/setting/setting.go:832:NewContext() [I] Default UI Location is Europe/Vienna
2020/10/26 09:54:39 ...dules/setting/git.go:91:newGit() [I] Git Version: 2.29.1.341.ge80a0c044a, Wire Protocol Version 2 Enabled
Proceed with sending email? [Y/n] y
Succseded: Was sent <!DOCTYPE html>
<html lang="en-US" class="theme-">
<head data-suburl="">
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta http-equiv="x-ua-compatible" content="ie=edge">
	<title>Internal Server Error -  git.MYDOMAIN.TLD </title>
	<link rel="manifest" href="/manifest.json" crossorigin="use-credentials">
	<meta name="theme-color" content="#782f59">
	<meta name="author" content="MYDOMAIN.TLD" />
	<meta name="description" content="git.MYDOMAIN.TLD Gitea instance by MYDOMAIN.TLD provided for fun as a hobby, with passion and love"/>
	<meta name="keywords" content="go,git,self-hosted,libre,golang,gitea">
	<meta name="referrer" content="no-referrer" />
	<meta name="_csrf" content="o9fyjFVqaM4GiwIOozs9nVA3sYE6MTYwMzcwMjQ4MzY1MTQ2MDI5MA" />




	<script>
		window.config = {
			AppVer: '1.14.0\u002bdev-91-g245597ab8',
			AppSubUrl: '',
			StaticUrlPrefix: '',
			UseServiceWorker:  true ,
			csrf: 'o9fyjFVqaM4GiwIOozs9nVA3sYE6MTYwMzcwMjQ4MzY1MTQ2MDI5MA',
			HighlightJS: false,
			SimpleMDE: false,
			Tribute: false,
			U2F: false,
			Heatmap: false,
			heatmapUser: null,
			NotificationSettings: {
				MinTimeout:  20000 ,
				TimeoutStep:   10000 ,
				MaxTimeout:  60000 ,
				EventSourceUpdateTime:  10000 ,
			},
			PageIsProjects: false,

		};
	</script>
	<link rel="icon" href="/img/favicon.svg" type="image/svg+xml">
	<link rel="alternate icon" href="/img/favicon.png" type="image/png">
	<link rel="mask-icon" href="/img/gitea-safari.svg" color="#609926">
	<link rel="fluid-icon" href="/img/gitea-lg.png" title="git.MYDOMAIN.TLD">

	<link rel="stylesheet" href="/css/index.css?v=ecd0cb9ecb8ba3904b86673f6741152e">
	<noscript>
		<style>
			.dropdown:hover > .menu { display: block; }
			.ui.secondary.menu .dropdown.item > .menu { margin-top: 0; }
		</style>
	</noscript>
	<style class="list-search-style"></style>

	<meta property="og:title" content="git.MYDOMAIN.TLD">
	<meta property="og:type" content="website" />
	<meta property="og:image" content="/img/gitea-lg.png" />
	<meta property="og:url" content="https://git.MYDOMAIN.TLD/" />
	<meta property="og:description" content="git.MYDOMAIN.TLD Gitea instance by MYDOMAIN.TLD provided for fun as a hobby, with passion and love">

<meta property="og:site_name" content="git.MYDOMAIN.TLD" />

	<link rel="stylesheet" href="/css/theme-42l-light.css?v=ecd0cb9ecb8ba3904b86673f6741152e">


</head>
<body>


	<div class="full height">
		<noscript>This website works better with JavaScript.</noscript>




			<div class="ui top secondary stackable main menu following bar light">
				<div class="ui container" id="navbar">
	<div class="item brand" style="justify-content: space-between;">
		<a href="/">
			<img class="ui mini image" src="/img/git.MYDOMAIN.TLD.svg" alt="git.MYDOMAIN.TLD icon">
		</a>
		<div class="ui basic icon button mobile-only" id="navbar-expand-toggle">
			<i class="sidebar icon"></i>
		</div>
	</div>

		<a class="item " href="/">Home</a>
		<a class="item " href="/explore/repos">Explore</a>





		<a class="item" target="_blank" rel="noopener noreferrer" href="https://docs.gitea.io">Help</a>
		<div class="right stackable menu">

				<a class="item" href="/user/sign_up">
					<svg viewBox="0 0 16 16" class="svg octicon-person" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M10.5 5a2.5 2.5 0 11-5 0 2.5 2.5 0 015 0zm.061 3.073a4 4 0 10-5.123 0 6.004 6.004 0 00-3.431 5.142.75.750 001.498.07 4.5 4.5 0 018.99 0 .75.75 0 101.498-.07 6.005 6.005 0 00-3.432-5.142z"/></svg> Register
				</a>

			<a class="item" rel="nofollow" href="/user/login?redirect_to=%2fapi%2finternal%2fmail%2fsend">
				<svg viewBox="0 0 16 16" class="svg octicon-sign-in" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M2 2.75C2 1.784 2.784 1 3.75 1h2.5a.75.75 0 010 1.5h-2.5a.25.25 0 00-.25.25v10.5c0 .138.112.25.25.25h2.5a.75.750 010 1.5h-2.5A1.75 1.75 0 012 13.25V2.75zm6.56 4.5l1.97-1.97a.75.75 0 10-1.06-1.06L6.22 7.47a.75.75 0 000 1.06l3.25 3.25a.75.75 0 101.06-1.06L8.56 8.75h5.69a.75.75 0 000-1.5H8.56z"/></svg> Sign In
			</a>
		</div>

</div>

			</div>



<div class="ui container center">
	<p style="margin-top: 100px"><img class="ui centered image" src="/img/500.png" alt="500"/></p>
	<div class="ui divider"></div>
	<br>

	<p>Gitea Version: 1.14.0&#43;dev-91-g245597ab8</p>

</div>




	</div>



	<footer>
	<div class="ui container">
		<div class="ui left">
			Powered by Gitea Version: 1.14.0&#43;dev-91-g245597ab8 Page: <strong>4ms</strong> Template: <strong>0ms</strong>
		</div>
		<div class="ui right links">

			<div class="ui language bottom floating slide up dropdown link item">
				<svg viewBox="0 0 16 16" class="svg octicon-globe" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M1.543 7.25h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.506 6.506 0 00-4.666 5.5zm2.733 1.5H1.543a6.506 6.506 0 004.666 5.5 11.13 11.13 0 01-.352-.552c-.715-1.192-1.437-2.874-1.581-4.948zm1.504 0h4.44a9.637 9.637 0 01-1.363 4.177c-.306.51-.612.919-.857 1.215a9.978 9.978 0 01-.857-1.215A9.637 9.637 0 015.78 8.75zm4.44-1.5H5.78a9.637 9.637 0 011.363-4.177c.306-.51.612-.919.857-1.215.245.296.55.705.857 1.215A9.638 9.638 0 0110.22 7.25zm1.504 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.506 6.506 0 004.666-5.5h-2.733zm2.733-1.5h-2.733c-.144-2.074-.866-3.756-1.58-4.948a11.738 11.738 0 00-.353-.552 6.506 6.506 0 014.666 5.5zM8 0a8 8 0 100 16A8 8 0 008 0z"/></svg>
				<div class="text">en</div>
				<div class="menu">

						<a lang="en-US" class="item active selected" href="#">en</a>

				</div>
			</div>
			<a href="/js/licenses.txt">Licenses</a>
			<a href="/api/swagger">API</a>
			<a href="/ssh-hostkeys.txt" rel="noreferer">hostkeys</a>
<a href="https://status.MYDOMAIN.TLD/" rel="noopener">Status</a>
<a href="/MYDOMAIN.TLD" rel="noopener">&#183; MYDOMAIN.TLD &#183;</a>
<a href="https://MYDOMAIN.TLD/contact/" rel="noopener">Contact</a>
<a href="/0DD13DBC6B5433D3.asc" title="default signing key" rel="noopener">0DD13DBC6B5433D3</a>

			<span class="version">Go1.15.3</span>
		</div>
	</div>
</footer>
```
There are some 500s in the chunk above and the email hasn't arrived by now (test email from the admin dash arrives immediately) so I assume something's not right here.

edits: typo and link to the patch

**Comment by lunny** [comment]
@wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf Thanks for your feedback, I have sent another commits, could you help to try again?

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
> @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf Thanks for your feedback, I have sent another commits, could you help to try again?

Sure and thank *you* for the commits.
Give me a minute, please.

**Review by mrsdizzie — APPROVED** [review]

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
Alright the emails do arrive now 🚀

File: modules/private/mail.go [hunk 1]
```diff
@@ -49,5 +49,10 @@ func SendEmail(subject, message string, to []string) (int, string) {
 		return http.StatusInternalServerError, fmt.Sprintf("Response body error: %v", err.Error())
 	}
 
-	return http.StatusOK, fmt.Sprintf("Was sent %s from %d", body, len(to))
+	var users = fmt.Sprintf("%d", len(to))
+	if len(to) == 0 {
+		users = "all"
+	}
+
+	return http.StatusOK, fmt.Sprintf("Was sent %s from %s", body, users)
```
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [inline modules/private/mail.go]: I get this as a response
```log
Success: Was sent 12 from all
```
image
which maybe should have been something like `"Sent 12 emails in total to all users"`

Comment (techknowlogick) [reply modules/private/mail.go]:
```suggestion
	return http.StatusOK, fmt.Sprintf("Sent %s email(s) to %s users", body, users)
```

Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply modules/private/mail.go]: yep, this...cheers @techknowlogick @lunny

**Pull request merged**